packages/trie/src/trie.ts

```diff
@@ -43,6 +43,8 @@ import type {
 import type { OnFound } from './util/asyncWalk.js'
 import type { BatchDBOp, DB, PutBatch } from '@ethereumjs/util'
 import type { Debugger } from 'debug'
+// eslint-disable-next-line implicit-dependencies/no-implicit
+import type { ReadableStream } from 'node:stream/web'
```

Comment:
Maybe also for others to chime in (@jochem-brouwer @acolytec3, everyone): will this type import affect browser and or other runtime support (bun, deno) in any way?

We generally have the policy to not add Node.js specific package imports. Does this also go for types? I am a bit unsure here. For web it shouldn't make a difference since web is using JavaScript (?). For bun I tested a trie package test run with `bunx vitest run test`. This does work, not sure if vitest "shields" something here.

I am totally undecided. Would it eventually make sense to spare out this import and rather go for `any`? Or are we on the safe side here?

Reply:
I am no browser expert here but I would assume that bun / deno do not have access to this `node:stream/web` package, however your `bunx vitest run test` indeed seems to pass. However I would assume that this would fail on browser builds, which does not seem to be the case (and the CI browser tests also pass). I have tried `bunx vitest run --config=./vitest.config.browser.ts --browser.name=chrome --browser.headless`, this fails all tests:

`Error: Module "events" has been externalized for browser compatibility. Cannot access "events.EventEmitter" in client code.  See https://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
`

However this does not seem to be related to `ReadableStream` :thinking:

Reply:
The point here is: since this is a web API, this import should just not be needed in the browser, and the code work nevertheless. I’m am just not sure how these cases are handled in practice. It would (eventually?) e.g. make sense if browsers just „overlook“ (aka: ignore) such an import.

In this case it is even a type import which should go away for JavaScript anyhow (re-read my message above). So generally I would assume we are on the safe side here.

@scorbajio maybe do add a comment above this import shortly explaining why this can go as an exception and stating that we normally do not do Node.js imports in our libs (not that this serves other devs as a reference to conclude: Node.js allowed). A date for the comment also can’t hurt to get some context.

(and the alternative is still that we go for „any“ here, might also be a justified choice in this case)

Reply:
I added a note detailing the choice to deviate from our import policy and kept the typing.

Reply:
Thanks, great comment! 🙏

Reply:
Also have added to the deprecation issue #3216 as well

Reply:
Hello @scorbajio @holgerd77 , we current have an issue building for the browser using webpack:
```
ERROR in node:stream/web
Module build failed: UnhandledSchemeError: Reading from "node:stream/web" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs.
```
could this be related to this change and if yes what would you recommend to make it build for the browser?
Thanks very much!

Reply:
@yann300 ah, we were arguing if this might bite us or not 😬 😛 (as seen above), seems we now have the confirmation.

@scorbajio can you prepare a short PR where we replace this with `any` and remove the import, as discussed above? That would be nice!

@yann300 Guess we'll make some bugfix releases relatively soon, maybe give us a couple more days, there might be 2-3 other things we would want to include as well.

Reply:
I've raised #3280 to address this.

Reply:
@yann300 Would you be able to provide the steps to reproduce this webpack build error so we can explore the problem and potential fixes?